taegis_magic/pandas/assets.py

```diff
@@ -0,0 +1,166 @@
+"""Pandas functions for Asset Lookups in Event and Alert DataFrames."""
+
+import logging
+import pandas as pd
+from typing import List, Optional
+from dataclasses import asdict
+
+from taegis_sdk_python import GraphQLService
+from taegis_sdk_python.services.assets2.types import (
+    AssetV2,
+    AssetFilter,
+    AssetWhereInputV2,
+)
+from taegis_magic.pandas.utils import chunk_list, get_tenant_id
+from taegis_magic.core.service import get_service
+
+log = logging.getLogger(__name__)
+
+
+def get_assets(
+    service: GraphQLService,
+    filter: AssetFilter,
+) -> List[AssetV2]:
+    """Takes a asset filter to obtain a list of assets
+
+    Parameters
+    ----------
+    service : GraphQLService
+        Taegis SDK GraphQL service.
+    filter : AssetFilter
+        Asset filter that specifies what assets to return back.
+
+    Returns
+    -------
+    List[AssetV2]
+        Returns a list of AssetV2 objects.
+    """
+    all_assets = []
+
+    try:
+        response = service.assets2.query.assets_v2(
+            filter_=filter,
+        )
+    except Exception as exc:
+        log.error(exc)
```

Comment:
Do we want to return early here if the exception is thrown? Can we guarantee that there will be a `response` object or want to do pagination over it?